questions/110-medium-capitalize/template.ts

```diff
@@ -1 +1,36 @@
-export type Capitalize<S extends string> = any
+const ABC = {
+  a: 'A',
+  b: 'B',
+  c: 'C',
+  d: 'D',
+  e: 'E',
+  f: 'F',
+  g: 'G',
+  h: 'H',
+  i: 'I',
+  j: 'J',
+  k: 'K',
+  l: 'L',
+  m: 'M',
+  n: 'N',
+  o: 'O',
+  p: 'P',
+  q: 'Q',
+  r: 'R',
+  s: 'S',
+  t: 'T',
+  u: 'U',
+  v: 'V',
+  w: 'W',
+  x: 'X',
+  y: 'Y',
+  z: 'Z',
+} as const
+
+export type Capitalize<
+  S extends string
+> = S extends `${infer First}${infer Rest}`
+  ? First extends keyof typeof ABC
+    ? `${typeof ABC[First]}${Rest}`
+    : S
+  : S
```

Comment:
I recommend you use Uppercase, hardcode the letters works but is not a scalable solution.
```suggestion
export type Capitalize<S extends string> = S extends `${infer First}${infer Rest}`
  ? `${Uppercase<First>}${Rest}`
  : S;
```